e2e/testcases/override_reconcile_timeout_test.go

```diff
@@ -122,7 +131,7 @@ func TestOverrideReconcileTimeout(t *testing.T) {
 	} else {
 		expectActuationStatus := "Succeeded"
 		expectReconcileStatus := "Succeeded"
-		err = verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus)
+		_, err := nomostest.Retry(nt.DefaultWaitTimeout, func() error { return verifyObjectStatus(u, idToVerify, expectActuationStatus, expectReconcileStatus) })
```

Comment:
This retry doesn't make any sense. The input is always the same object that was retrieved in the previous Get.

If you want to do this right, I suggest converting it to use WatchForObject and then make a new Precondition factory function that returns a Precondition that checks the ResourceGroup status.resourceStatuses contents. Re-writing  verifyObjectStatus to use a ResourceGroup instead of an Unstructured will make it simpler to read too.

Reply:
Sure, changed to use WatchForObject and use ResourGroup in Precondition function